src/services/refactors/extractMethod.ts

```diff
@@ -0,0 +1,885 @@
+            changes: changeTracker.getChanges()
+        };
+
+        function getPropertyAssignmentsForWrites(writes: UsageEntry[]) {
+            return writes.map(w => createShorthandPropertyAssignment(w.symbol.name));
+        }
+
+        function generateReturnValueProperty() {
+            return "__return";
+        }
+
+        function transformFunctionBody(n: Node) {
+            if (isBlock(n) && !writes && substitutions.size === 0) {
+                // already block, no writes to propagate back, no substitutions - can use node as is
+                return { body: n, returnValueProperty: undefined };
+            }
+            let returnValueProperty: string;
+            const statements = createNodeArray(isBlock(n) ? n.statements.slice(0) : [isStatement(n) ? n : createReturn(<Expression>n)]);
+            // rewrite body if either there are writes that should be propagated back via return statements or there are substitutions
+            if (writes || substitutions.size) {
+                const rewrittenStatements = visitNodes(statements, visitor);
+                if (writes && !(range.facts & RangeFacts.HasReturn)) {
+                    // add return at the end to propagate writes back in case if control flow falls out of the function body
+                    // it is ok to know that range has at least one return since it we only allow unconditional returns
+                    rewrittenStatements.push(createReturn(createObjectLiteral(getPropertyAssignmentsForWrites(writes))));
+                }
+                return { body: createBlock(rewrittenStatements), returnValueProperty: returnValueProperty };
+            }
+            else {
+                return { body: createBlock(statements), returnValueProperty: undefined };
+            }
+
+            function visitor(node: Node): VisitResult<Node> {
+                if (node.kind === SyntaxKind.ReturnStatement && writes) {
+                    const assignments: ObjectLiteralElementLike[] = getPropertyAssignmentsForWrites(writes);
+                    if ((<ReturnStatement>node).expression) {
+                        if (!returnValueProperty) {
+                            returnValueProperty = generateReturnValueProperty();
+                        }
+                        assignments.push(createPropertyAssignment(returnValueProperty, visitNode((<ReturnStatement>node).expression, visitor)));
+                    }
+                    return createReturn(createObjectLiteral(assignments));
+                }
+                else {
+                    const substitution = substitutions.get(getNodeId(node).toString());
+                    return substitution || visitEachChild(node, visitor, nullTransformationContext);
+                }
+            }
+        }
+    }
+
+    const nullTransformationContext: TransformationContext = {
+        enableEmitNotification: noop,
+        enableSubstitution: noop,
+        endLexicalEnvironment: () => undefined,

```

Comment:
Sort of. `noop` returns `void`, but `endLexicalEnvironment` expects a `Statement[] | undefined` return value. Switched to `noop` and added a type assertion